controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +132,54 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 	return response, nil
 }
 
+// RemoveEtcdMissingNodes iterates over all ETCD members and finds members with nodes that do not belong to the workload cluster.
+// If there are any such members, it deletes those members and remove nodes from kubeadmconfig so that Kubeadm does not try to run etcd-health check over them.
+func (w *Workload) RemoveEtcdMissingNodes(ctx context.Context) error {
+	controlPlaneNodes, err := w.getControlPlaneNodes(ctx)
+	if err != nil {
+		return err
+	}
+
+	errList := []error{}
+	for _, node := range controlPlaneNodes.Items {
```

Comment:
Do we need to do this as a triple nested loop? I think we can just connect to the etcd leader and only need to call `Members` once.

Reply:
This code is working right now it breaks the whole cluster. I think it just goes through every node and deletes the etcd member for it.

Reply:
Yeah, broke this while fixing another comment above. Good catch. Adding some e2e tests to catch these.